pkg/sass-parser/lib/src/expression/function.test.ts

```diff
@@ -0,0 +1,316 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {ArgumentList, FunctionExpression} from '../..';
+import * as utils from '../../../test/utils';
+
+describe('a function expression', () => {
+  let node: FunctionExpression;
+
+  describe('with no namespace', () => {
+    function describeNode(
+      description: string,
+      create: () => FunctionExpression,
+    ): void {
+      describe(description, () => {
+        beforeEach(() => void (node = create()));
+
+        it('has sassType function-call', () =>
+          expect(node.sassType).toBe('function-call'));
+
+        it('has no namespace', () => expect(node.namespace).toBe(undefined));
+
+        it('has a name', () => expect(node.name).toBe('foo'));
+
+        it('has an argument', () =>
+          expect(node.arguments.nodes[0]).toHaveStringExpression(
+            'value',
+            'bar',
+          ));
+      });
+    }
+
+    describeNode('parsed', () => utils.parseExpression('foo(bar)'));
+
+    describeNode(
+      'constructed manually',
+      () => new FunctionExpression({name: 'foo', arguments: [{text: 'bar'}]}),
+    );
+
+    describeNode('constructed from ExpressionProps', () =>
+      utils.fromExpressionProps({name: 'foo', arguments: [{text: 'bar'}]}),
+    );
+  });
+
+  describe('with a namespace', () => {
+    function describeNode(
+      description: string,
+      create: () => FunctionExpression,
+    ): void {
+      describe(description, () => {
+        beforeEach(() => void (node = create()));
+
+        it('has sassType function-call', () =>
+          expect(node.sassType).toBe('function-call'));
+
+        it('has a namespace', () => expect(node.namespace).toBe('baz'));
+
+        it('has a name', () => expect(node.name).toBe('foo'));
+
+        it('has an argument', () =>
+          expect(node.arguments.nodes[0]).toHaveStringExpression(
+            'value',
+            'bar',
+          ));
+      });
+    }
+
+    describeNode('parsed', () => utils.parseExpression('baz.foo(bar)'));
+
+    describeNode(
+      'constructed manually',
+      () =>
+        new FunctionExpression({
+          namespace: 'baz',
+          name: 'foo',
+          arguments: [{text: 'bar'}],
+        }),
+    );
+
+    describeNode('constructed from ExpressionProps', () =>
+      utils.fromExpressionProps({
+        namespace: 'baz',
+        name: 'foo',
+        arguments: [{text: 'bar'}],
+      }),
+    );
+  });
+
+  describe('if()', () => {
+    beforeEach(() => void (node = utils.parseExpression('if(cond, bar, baz)')));
```

Comment:
TIL `if()` is a function in Sass! 🤯

Reply:
Only sort of! It's actually technically its own syntactic construct, distinct from a standard function in that it only ever evaluates one of the `true` or `false` branches. Honestly at this point we could just make it check that at runtime like we do for `calc()` _et al_, but we haven't made that change yet.